pkg/sql/cacheutil/cache.go

```diff
@@ -28,6 +29,7 @@ import (
 // during user authentication and session initialization.
 type Cache struct {
 	syncutil.Mutex
+	WarmingGroup  *sync.WaitGroup
```

Comment:
The problem with WaitGroup is that it doesn't compose with context cancellation. It also feels a little weird to me to attach it to `cacheutil.Cache` when this is the only one

Reply:
my thought was that any cache should be able to implement cache warming logic

Reply:
That's fair enough, but I still think there's lots of value in wrapping the generic mechanism even if it's a thin wrapper.